src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
+      p.p = fftw_c.fftw_plan_many_dft(dim-1, &n_batch[0], n[dim-1], [&fftw_c.fftw_complex](input_base), &n_batch[0], 1, i_dist, [&fftw_c.fftw_complex](output_base), &n_batch[0], 1, i_dist,  fftw_c.FFTW_FORWARD, fftw_c.FFTW_ESTIMATE)
+
+    end
     p.address_space = address_space
-    ;[(function()
-         if use_cuda then
-           return rquote
-             if iface.get_num_local_gpus() > 0 then
-               p.cufft_p = make_plan_gpu(input, output, address_space)
-             end
-           end
-         else
-           return rquote end
-         end
-       end)()]
   end
 
-  task iface.make_plan_task(input : region(ispace(itype), dtype),
-                            output : region(ispace(itype), dtype),
-                            plan : region(ispace(int1d), iface.plan))
+  --The make_plan tasks above are demand_(__inline tasks). This means that if the user wants it to execute it in a separate task, they must wrap the task themselves -- this is what make_plan_task does.
+  task iface.make_plan_task(input : region(ispace(itype), dtype_in), output : region(ispace(itype), dtype_out), plan : region(ispace(int1d), iface.plan))
   where reads writes(input, output, plan) do
     iface.make_plan(input, output, plan)
   end
 
+  --Distribution version of make_plan 
   __demand(__inline)
-  task iface.make_plan_distrib(input : region(ispace(itype), dtype),
-                               input_part : partition(disjoint, input, ispace(int1d)),
-                               output : region(ispace(itype), dtype),
-                               output_part : partition(disjoint, output, ispace(int1d)),
-                               plan : region(ispace(int1d), iface.plan),
-                               plan_part : partition(disjoint, plan, ispace(int1d)))
+  task iface.make_plan_distrib(input : region(ispace(itype), dtype_in), input_part : partition(disjoint, input, ispace(int1d)), output : region(ispace(itype), dtype_out), output_part : partition(disjoint, output, ispace(int1d)), plan : region(ispace(int1d), iface.plan), plan_part : partition(disjoint, plan, ispace(int1d)))
   where reads writes(input, output, plan) do
+    
+    --Get number of nodes and check consistency of nodes/colors
     var n = iface.get_num_nodes()
     regentlib.assert(input_part.colors.bounds.hi - input_part.colors.bounds.lo + 1 == int1d(n), "input_part colors size must be equal to the number of nodes")
     regentlib.assert(input_part.colors.bounds == output_part.colors.bounds, "input_part and output_part colors must be equal")
     regentlib.assert(input_part.colors.bounds == plan_part.colors.bounds, "input_part and plan_part colors must be equal")
 
     var p : iface.plan
+    --T(x) is a cast from type T to a value x
     p.p = [fftw_c.fftw_plan](0)
-    p.address_space
-    ;[(function() if use_cuda then return rquote p.cufft_p = 0 end else return rquote end end end)()]
+
+    if gpu_available then
+      p.cufft_p = 0
+    end
```

Comment:
I think you can't just do this. The field does not even exist in the non-GPU case.

Reply:
I think this is resolved by the switch to rescape / remit rquote and the breaking into a separate execute_plan_gpu, but let me know if not